core/src/apps/management/reset_device.py

```diff
@@ -112,8 +114,10 @@
 
 
 def _validate_reset_device(msg: ResetDevice) -> None:
+    if msg.backup_type == ResetDeviceType.Slip39_Multiple_Groups:
```

Comment:
If msg.backup_type is > 2, then I think the code silently defaults to BIP-0039. So I would prefer to see something like:
```
if msg.backup_type not in (ResetDeviceType.Bip39, ResetDeviceType.Slip39_Single_Group):
    raise wire.ProcessError("Backup type not implemented.")
```

Reply:
Hm, okay, but we should also allow None. Because older implementations do not know about this field. @matejcik any comments on this?

Reply:
what happens if you, say, use old trezorctl and the device starts a Shamir backup? will it work? do we want that?

Reply:
Mmm. You can't intialize Shamir wallet with old trezorctl, becuase the backup type field is not there.

You could intialize Shamir wallet with new trezorctl and then move to old trezorctl and do the backup. That could work probably. But why?